fbw-common/src/wasm/systems/systems/src/surveillance/egpws/runtime.rs

```diff
@@ -0,0 +1,1286 @@
+
+            mode_1_sinkrate_lamp_active: false,
+            mode_1_sinkrate_voice_active: false,
+            mode_1_pull_up_active: false,
+
+            mode_2_takeoff_latch_node: MonostableTriggerNode::new_falling(Duration::from_secs(60)),
+            mode_2_closure_rate_derivative: DerivativeNode::new(),
+            mode_2_ra_rate_limiter: RateLimiter::new_symmetrical(170.), // Roughly 10 kfpm, i.e. maximum of boundary
+            mode_2_closure_rate_rate_limiter: RateLimiter::new_symmetrical(2_900.), // roughly 1.5g in ft/(min*s)
+            mode_2_ra_filter: LowPassFilter::new(Duration::from_secs_f64(1.)),
+
+            mode_2_pull_up_preface_voice_emitted: false,
+            mode_2_pull_up_preface_active: false,
+            mode_2_pull_up_active: false,
+            mode_2_terrain_active: false,
+
+            mode_3_max_achieved_alt_ft: None,
+            mode_3_declutter_threshold_increase: 0.,
+
+            mode_3_lamp_active: false,
+            mode_3_dont_sink_voice_active: false,
+
+            mode_4_c_filter_value: 0.,
+            mode_4_a_alert_upper_boundary: 500.,
+            mode_4_c_declutter_threshold_increase: 0.,
+            mode_4_b_active: false,
+            mode_4_ab_declutter_threshold_increase: 0.,
+            mode_4_ab_prev_submode_active: 0,
+            mode_4_too_low_gear_voice_active: false,
+            mode_4_too_low_flaps_voice_active: false,
+            mode_4_too_low_terrain_voice_active: false,
+            mode_4_lamp_active: false,
+
+            mode_5_cancel_active: false,
+            mode_5_time_to_next_aural: Duration::ZERO,
+            mode_5_declutter_threshold_increase: 0.,
+            mode_5_glideslope_soft_voice_active: false,
+            mode_5_glideslope_hard_voice_active: false,
+            mode_5_glideslope_lamp_active: false,
+
+            number_of_aural_warning_emissions: 0,
+            time_since_first_emission: Duration::ZERO,
+
+            warning_lamp_activated: false,
+            alert_lamp_activated: false,
+            aural_output: AuralWarning::None,
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        discrete_inputs: &TerrainAwarenessWarningSystemDiscreteInputs,
+        ra1: &impl RadioAltimeter,
+        ra2: &impl RadioAltimeter,
+        adr: &impl AirDataReferenceBus,
+        ir: &impl InertialReferenceBus,
+        ils: &impl InstrumentLandingSystemBus,
+    ) {
+        // First, check if we're still starting up and if so, simulate a wait until all self tests
+        // have completed.
+        if let Some(new_remaining) = self.remaining_startup.checked_sub(context.delta()) {
+            self.remaining_startup = new_remaining;
+        } else {
+            self.remaining_startup = Duration::ZERO;
+        }
+
+        // If there's any startup time remaining, do nothing
+        if self.remaining_startup > Duration::ZERO {
```

Comment:
How about using the helper function?

```suggestion
        if !self.is_initialized() {
```